homeassistant/components/xiaomi_miio/select.py

```diff
@@ -0,0 +1,258 @@
+}
+
+
+@dataclass
+class SelectorType:
+    """Class that holds device specific info for a xiaomi aqara or humidifier selectors."""
+
+    name: str = None
+    short_name: str = None
+    unit_of_measurement: str = None
+    icon: str = None
+    device_class: str = None
+    min: float = None
+    max: float = None
+    mode: str = None
+    options: list = None
+    step: float = None
+    service: str = None
+
+
+SELECTOR_TYPES = {
+    FEATURE_SET_LED_BRIGHTNESS: SelectorType(
+        name="Led brightness",
+        icon="mdi:brightness-6",
+        short_name=ATTR_LED_BRIGHTNESS,
+        options=["Bright", "Dim", "Off"],
+        service=SERVICE_SET_LED_BRIGHTNESS,
+    ),
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Import Miio configuration from YAML."""
+    _LOGGER.warning(
+        "Loading Xiaomi Miio Fan via platform setup is deprecated. "
+        "Please remove it from your configuration"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Selectors from a config entry."""
+    entities = []
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+        host = config_entry.data[CONF_HOST]
+        token = config_entry.data[CONF_TOKEN]
+        model = config_entry.data[CONF_MODEL]
+        device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+        coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+        device_features = 0
+        entity_class = None
+
+        _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+
+        if model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB
+            entity_class = XiaomiAirHumidifierSelector
+        elif model in [MODEL_AIRHUMIDIFIER_CA4]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA4
+            entity_class = XiaomiAirHumidifierMiotSelector
+        elif model in MODELS_HUMIDIFIER:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER
+            entity_class = XiaomiAirHumidifierSelector
+        else:
+            return
+
+        for feature in SELECTOR_TYPES:
+            selector = SELECTOR_TYPES[feature]
+            if feature & device_features and feature in SELECTOR_TYPES:
+                entities.append(
+                    entity_class(
+                        f"{config_entry.title} {selector.name}",
+                        device,
+                        config_entry,
+                        f"{selector.short_name}_{config_entry.unique_id}",
+                        selector,
+                        coordinator,
+                    )
+                )
+
+    async_add_entities(entities, update_before_add=True)
```

Comment:
Do we need `update_before_add=True`? We have data from the device because of [this](https://github.com/home-assistant/core/blob/bf35adccec6b89ff0db5dc0f815295f2c76330fc/homeassistant/components/xiaomi_miio/__init__.py#L140)

Reply:
Apparently we do
![afbeelding](https://user-images.githubusercontent.com/7188918/123121503-812d6580-d445-11eb-8fc1-2eff6a558f9d.png)

I think this is because we still need the entities to trigger to update. The device data is fetched, but:
```
@callback
def _handle_coordinator_update(self)
```
Is not called from the entities, because they do not yet exist.